### DISCUSSION TITLE
sqlite3.h

### DISCUSSION THREAD

**Discussion by digitalProjectsOfFuture** [virtual-maker/VirtualBoard#8] [General]
Can I write a project with <sqlite3.h> using virtualboard?

**Comment by virtual-maker** [comment]
@digitalProjectsOfFuture
Yes, in basic you can include additional Arduino libraries in your VirtualBoard project. 
But these libraries have to be compatible with MSVC (Microsoft Visual C++) or you have to create a special MSVC variant of the library.

In your special case you can try to include the library [Sqlite3 Arduino library for ESP32](https://github.com/siara-cc/esp32_arduino_sqlite3_lib).

An additional problem in your case are the libraries to access an SD card.

```
#include <FS.h>
#include "SPIFFS.h"
#include "SD_MMC.h"
#include "SD.h"
```
These also need to be adapted to MSVC first. 

I also think it would be useful to simulate the SD card.
So to use the filesystem of Windows when the Arduino sketch calls the SD functions.
A subfolder in your project could then simulate the SD card.

So all in all there is still a lot of work to do to include sqlite3.h.

BR Immo
